README.md

```diff
@@ -1 +1,97 @@
-# RL-based-Maze-Solver
\ No newline at end of file
+# RL-based Maze Solver
+Open Projects 2021
+
+![Input Maze Solution](https://github.com/tushdon2/RL-based-Maze-Solver/blob/main/Images%20and%20Videos/Images/Input%20maze1%20and%20solution.png)
+<br>fig1 <i>Input Maze Solution</i>
+
+![Random Maze Solution](https://github.com/tushdon2/RL-based-Maze-Solver/blob/main/Images%20and%20Videos/Images/Randomly%20generated%20maze%20and%20Solution.png)
+<br>fig2 <i>Random Maze Solution</i>
+
+## Abstract
+<p align = "justify">The aim of this project is to solve a given 2D maze (given as an image input) using RL (Q-Learning) algorithm. The program first processes the image input of the maze and converts it into a matrix. The Q-Learning algorithm then uses this matrix to generate a q-matrix, which is finally used to get the shortest path. The program can also generate random mazes and then solve it.</p>
+
+## Motivation
+<p align = "justify">Reinforcement learning in robotics has been a challenging domain in the field of AI for the past few years. The ability to equip a robot with a powerful enough tool to allow an autonomous discovery of an optimal behavior through trial-and-error interactions with its environment has paved the path to many deep research projects.<br></br>
+Our project inspires us to solve many problems, like Autonomous Mobile Robot Obstacle Prevention using Q-learning.</p>
+
+![Image Processing](https://github.com/tushdon2/RL-based-Maze-Solver/blob/main/Images%20and%20Videos/Images/Input%20maze2_image%20processing.png)
+<br>fig3 <i>Image Processing</i>
+
+<!-- ## Mechanical Aspect of the Design -->
+
+<!-- ## Electronics Aspect of the Design -->
+
+## Software Aspect of the Design

```

Comment:
Justify the text for remaining content like Software Aspects, Cost Structure, Limitations, Future Improvements etc which is currently missing.

Reply:
justified the paragraphs

Reply:
updated the README